provisioner/template.go

```
@@ -839,3 +840,9 @@ func scaleQuantity(quantityStr string, factor float32) (string, error) {
 	}
 	return quantity.String(), nil
 }
+
+// clusterName returns the canonical cluster name based on the cluster
+// provider.
+func clusterName(cluster api.Cluster) (string, error) {
```

This doesn't need to return an error.

---

I assume we can't use `cluster.Name()` in the templates directly?

---

Dropped this function as we can use `.Cluster.Name` in templates. The test is updated to use this.